### ISSUE TITLE
Suggestion: add clib package manager support

### ISSUE THREAD

**Issue by RaZeR-RBI** [xmake-io/xmake#397]
What do you think about adding support for [clib package manager](https://github.com/clibs/clib)? I've used it for a small project, seems very simple to use and install, but it doesn't provide any ways to compile the dependencies. It has a package.json definition which looks similar to npm one, which should be not very hard to parse.

I can try to implement it myself and send a PR if you don't mind :smiley:

**Comment by waruqi** [comment]
Yes, you can refer to https://github.com/xmake-io/xmake/tree/master/xmake/modules/package/manager/conan

Then implement modules `/package/manager/clib/find_package.lua` and install_package.lua   

And run  

```lua
cd projectdir
xmake l package.manager.find_package xxx
```
to test these module interfaces.

**Comment by waruqi** [comment]
Example:

```lua
add_requires("clib::clibs/bytes@0.0.4", {alias = "bytes"})

target("xmake-test")
    set_kind("binary")
    add_files("clib/bytes/*.c")
    add_files("src/*.c") 
    add_packages("bytes")
```
